crux-llvm/test-data/golden/copysign.c

```diff
@@ -0,0 +1,45 @@
+#include <crucible.h>
+#include <float.h>
+#include <math.h>
+#include <stdbool.h>
+
+// A reasonable specification for copysign, assuming that x isn't NAN or
+// INFINITY.
+float copysign_spec(float x, float y) {
+  float magnitude = fabs(x);
+  if (signbit(y)) {
+    return -magnitude;
+  } else {
+    return magnitude;
+  }
+}
+
+bool float_approx_equal(float x, float y) {
```

Comment:
For my curiosity, is this actually necessary here? Flipping the sign bit is one of the few things you can do with floating-point where you'll actually get the exactly equalities you expect.

Reply:
I used `FLT_EPSILON` due to my general paranoia about using `==` with floating-point values, but I think you're right in that I'm being excessively paranoid here. I've removed this in favor of plain old `==`.